lib/methods/veres-one/veres-did-doc.js

```diff
@@ -206,6 +206,106 @@ class VeresOneDidDoc {
     };
   }
 
+  /**
+   * Composes and returns a service id for a service name.
+   *
+   * @param serviceName {string}
+   *
+   * @returns {string} Service id
+   */
+  serviceIdFor(serviceName) {
+    if(!serviceName) {
+      throw new Error('Invalid service name');
+    }
+
+    return this.id + ';' + encodeURIComponent(serviceName);
+  }
+
+  /**
+   * Finds a service endpoint in this did doc, given an id or a name.
+   *
+   * @param [id] {string}
+   * @param [name] {string}
+   *
+   * @returns {object}
+   */
+  findService({id, name}) {
+    const jsonld = this.injector.use('jsonld');
+
+    const serviceId = id || this.serviceIdFor(name);
+
+    return jsonld
+      .getValues(this.doc, 'service')
+      .find(service => service.id === serviceId);
+  }
+
+  /**
+   * Tests whether this did doc has a service endpoint (by name or id).
+   * One of `id` or `name` is required.
+   *
+   * @param [id] {string}
+   * @param [name] {string}
+   *
+   * @returns {boolean}
+   */
+  hasService({id, name}) {
+    return !!this.findService({id, name});
+  }
+
+  /**
+   * Adds a service endpoint to this did doc.
+   * One of `id` or `name` is required.
+   *
+   * @param [id] {string}
+   * @param [name] {string}
+   * @param type {string} Endpoint type (e.g. 'AgentService')
+   * @param endpoint {string} Endpoint uri (e.g. 'https://agent.example.com')
+   * @param [options] {object} Any additional properties of endpoint
+   */
+  addService({id, name, type, endpoint, ...options}) {
+    const jsonld = this.injector.use('jsonld');
+
+    const serviceId = id || this.serviceIdFor(name);
+
+    if(!type) {
+      throw new Error('Service endpoint type is required');
+    }
+    if(!endpoint) {
+      throw new Error('Service endpoint uri is required');
+    }
+
+    jsonld.addValue(this.doc, 'service', {
```

Comment:
Do we need to be more precise here and augment/replace a service or throw an error if it already exists? I would think so.

Reply:
The quickest fix is to throw an error for a duplicate service, forcing a service to be removed prior to adding it with changes.

Reply:
Ah, I think the context was - did-client was doing a check for an existing service downstream, in client code. So I didn't include it in the 'addService()'. But that's probably a better place for it, yeah.

Reply:
K, should be fixed in latest commit.